base_layer/core/src/transaction.rs

```diff
@@ -319,12 +322,11 @@ impl TransactionKernel {
     pub fn verify_signature(&self) -> Result<(), TransactionError> {
         let excess = self.excess.as_public_key();
         let r = self.excess_sig.get_public_nonce();
-        let c = Challenge::<SignatureHash>::new()
-            .concat(r.as_bytes())
-            .concat(excess.as_bytes())
-            .concat(&self.fee.to_le_bytes())
-            .concat(&self.lock_height.to_le_bytes());
-
+        let m = TransactionMetadata {
+            lock_height: self.lock_height,
+            fee: self.fee,
+        };
+        let c = build_challenge(r, &m);
```

Comment:
should we not include the excess based on the meeting we had in the office?
We need to include it in the kernel publicly anyway and this will make the transactions more secure

Reply:
I'm fine to include it, but let's punt it to another PR